### PR TITLE
Functional tests for OData API api/v2/packages() caching 

### PR DESCRIPTION
**Pull request by skofman1**
1. APIM behavior:
- If a request meets the criteria it will be cached
- If a request meets the criteria and it's a cache miss we add header x-cache: MISS
- If a request meets the criteria and it's a cache hit we add header x-cache: HIT

2. The tests belong to a new Category (ApiManagement) and will NOT run normaly. In the future, when we automate APIM config deployment we can use them for verification. Until then, we will run them from VSTS manually during config deployment.

3. Changed the build to use MSBuild 15.0. Tested in VSTS for all tests.

### CODE REVIEW

File: tests/Scripts/RunTests.ps1 [hunk 1]
```diff
@@ -36,9 +36,12 @@ if ($LastExitCode) {
...
 $webTestsDirectory = "$rootName\NuGetGallery.WebUITests.$TestCategory\bin\$Config"
-& $msTest "/TestContainer:$webTestsDirectory\NuGetGallery.WebUITests.$TestCategory.dll" "/TestSettings:$rootName\Local.testsettings" "/detail:stdout" "/resultsfile:$webUITestResults"
-if ($LastExitCode) {
-    $exitCode = 1
+
+if(Test-Path $webTestsDirectory -PathType Container) { 
+	& $msTest "/TestContainer:$webTestsDirectory\NuGetGallery.WebUITests.$TestCategory.dll" "/TestSettings:$rootName\Local.testsettings" "/detail:stdout" "/resultsfile:$webUITestResults"
+	if ($LastExitCode) {
+		$exitCode = 1
+	}
```
Comment (loic-sharma) [inline tests/Scripts/RunTests.ps1]: >     [](start = 2, length = 4)

There some whitespace after the `}` here

File: tests/NuGetGallery.FunctionalTests/ApiManagement/CachingTests.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            // Arrange
+            string url = UrlHelper.V2FeedRootUrl + suffix;
+
+            Action<HttpResponseHeaders> verifyHeaders = (headers) => Assert.False(headers.Contains(CacheHeaderName));
+
+            // Act
+            var firstResponse = await MakeODataRequest(url);
+
+            // sleep for 5 seconds so the clock will shift
+            await Task.Delay(5000);
```
Comment (loic-sharma) [inline tests/NuGetGallery.FunctionalTests/ApiManagement/CachingTests.cs]: >5000 [](start = 29, length = 4)

Nit: `TimeSpan.FromSeconds(5)`

File: tests/NuGetGallery.FunctionalTests/ApiManagement/CachingTests.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            using (HttpResponseMessage response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, url)))
+            {
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+                response.Headers.TryGetValues(CacheHeaderName, out IEnumerable<string> cacheHeaders);
+
+                string content = await response.Content.ReadAsStringAsync();
+                using (var stringReader = new StringReader(content))
+                {
+                    using (var xmlReader = XmlReader.Create(stringReader))
```
Comment (loic-sharma) [inline tests/NuGetGallery.FunctionalTests/ApiManagement/CachingTests.cs]: >using (var xmlReader = XmlReader.Create(stringReader)) [](start = 20, length = 54)

Nit: This using can be on the same indentation as the previous using

File: tests/NuGetGallery.FunctionalTests/ApiManagement/CachingTests.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        [InlineData("Packages()?$skip=8356600&$top=80&$filter=(tolower(Id)%20eq%20'nlog')&$orderby=Id")]
+        [InlineData("Packages?IsLatestVersion%20and%20substringof('sample-data',Tags)&$top=250")]
+        [InlineData("Packages()?IsLatestVersion%20and%20substringof('sample-data',description)&$top=250")]
+        [Category("ApiManagementTests")]
+        public async Task RequestsThatShouldNotBeCachedAreNotCached(string suffix)
+        {
+            // Arrange
+            string url = UrlHelper.V2FeedRootUrl + suffix;
+
+            Action<HttpResponseHeaders> verifyHeaders = (headers) => Assert.False(headers.Contains(CacheHeaderName));
```
Comment (xavierdecoster) [inline tests/NuGetGallery.FunctionalTests/ApiManagement/CachingTests.cs]: Where is this action invoked?

Comment (skofman1) [reply tests/NuGetGallery.FunctionalTests/ApiManagement/CachingTests.cs]: forgot to remove :) thanks!

**Pull request merged**